### PR TITLE
feat(condo): DOMA-4351 added billing receipt notification deduplication logic

### PR DESCRIPTION
**Pull request by akarjakin**
Same user gets only one billing receipt notification at a time within each execution session of the script

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/property/utils/testSchema/index.js [hunk 1]
```diff
@@ -54,8 +54,8 @@ async function updateTestProperty (client, id, extraAttrs = {}) {
     return [obj, attrs]
 }
 
-async function makeClientWithProperty (includeFlat = false) {
-    const client = await makeClientWithRegisteredOrganization()
+async function makeClientWithProperty (includeFlat = false, clientWithOrganization) {
```
Comment (AleX83Xpert) [inline apps/condo/domains/property/utils/testSchema/index.js:57]: It's not good to add a mandatory parameter after the not mandatory one. If you want to keep the order, please add the default value.

File: apps/condo/domains/resident/tasks/helpers/spec.helpers.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ const { registerResidentByTestClient, registerServiceConsumerByTestClient } = re
 
 const { makeClientWithSupportUser, makeClientWithServiceUser, makeClientWithResidentUser } = require('@condo/domains/user/utils/testSchema')
 
-const makeBillingReceiptWithResident = async (billingReceiptAttrs = {}, skipConsumer = false) => {
-    const userClient = await makeClientWithProperty()
+const makeBillingReceiptWithResident = async (billingReceiptAttrs = {}, skipConsumer = false, clientWithOrganization) => {
```
Comment (AleX83Xpert) [inline apps/condo/domains/resident/tasks/helpers/spec.helpers.js:20]: default value

File: apps/condo/domains/property/utils/testSchema/index.js (hunk 1 above)
Comment (akarjakin) [reply apps/condo/domains/property/utils/testSchema/index.js:57]: it's not mandatory actually, you can freely skip it. it's only needed if you want to create BillingReceipt and all connected entities for same user

File: apps/condo/domains/resident/tasks/helpers/spec.helpers.js (hunk 2 above)
Comment (akarjakin) [reply apps/condo/domains/resident/tasks/helpers/spec.helpers.js:20]: it doesn't need default value

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/property/utils/testSchema/index.js (hunk 1 above)
Comment (akarjakin) [reply apps/condo/domains/property/utils/testSchema/index.js]: removed that

File: apps/condo/domains/resident/tasks/helpers/spec.helpers.js (hunk 2 above)
Comment (akarjakin) [reply apps/condo/domains/resident/tasks/helpers/spec.helpers.js]: removed that

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ const prepareAndSendNotification = async (context, receipt, resident) => {
 const sendBillingReceiptsAddedNotificationsForPeriod = async (receiptsWhere, onLastDtChange ) => {
     const { keystone: context } = await getSchemaCtx('Resident')
     const receiptsCount = await BillingReceipt.count(context, receiptsWhere)
-    let skip = 0, successCount = 0
+    let skip = 0, successCount = 0, notifiedUsers = {}
```
Comment (AleX83Xpert) [inline apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js:104]: Why don't use a `Set` (`add()` and `has()`)? It is faster and utilizes less memory.
```suggestion
    let skip = 0, successCount = 0
    const notifiedUsers = new Set()
```

Comment (akarjakin) [reply apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js]: fixed

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**